### PR TITLE
Fix clearstatcache on links

### PR DESCRIPTION
**Pull request by kimpepper**

### CODE REVIEW

File: src/SkprConfig.php [hunk 1]
```diff
@@ -41,7 +41,10 @@ public function load(string $filename = self::DEFAULT_FILENAME): self {
       return $this;
     }
     if (empty($this->config)) {
-      clearstatcache(TRUE, $filename);
+      while (is_link($filename) || is_link(dirname($filename))) {
+        clearstatcache(TRUE, $filename);
+        $filename = dirname($filename) . '/' . readlink($filename);
```
Comment (kimpepper) [inline src/SkprConfig.php]: @acbramley the dirname bit feels a bit weird. What if the link is absolute instead of relative?

Comment (kimpepper) [reply src/SkprConfig.php]: Actually, we don't really need to generalise here. We know exactly what the symlink chain will be on skpr

File: src/SkprConfig.php [hunk 2]
```diff
@@ -49,15 +49,15 @@ public function load(string $filename = self::DEFAULT_FILENAME, string $symlink
     }
     // We cache the config in memory.
     if (empty($this->config)) {
-      $data = @file_get_contents($filename);
+      $data = @file_get_contents(realpath($filename));
       // If the data is not found, symlinks may be outdated.
       if ($data === FALSE) {
```
Comment (acbramley) [inline src/SkprConfig.php:58]: Can't this just be an `if (!file_exists(realpath($filename)))` rather than trying to get the contents twice?

Comment (kimpepper) [reply src/SkprConfig.php:58]: If we get the contents we just jump to parsing the json and thats it. If the file doesn't exist, we clear caches and try again.

File: src/SkprConfig.php [hunk 3]
```diff
@@ -91,7 +133,10 @@ protected function convertToEnvVarName($key): string {
    * @return array
    *   Values.
    */
-  public function getAll(bool $environment_format = FALSE, string $filename = self::DEFAULT_FILENAME): array {
+  public function getAll(
+    bool $environment_format = FALSE,
+    string $filename = self::DEFAULT_FILENAME
+  ): array {
```
Comment (acbramley) [inline src/SkprConfig.php]: Why the new lines?

Comment (kimpepper) [reply src/SkprConfig.php]: autoformatting.

**Pull request merged**